lambda/src/main/scala/pricemigrationengine/model/ZuoraSubscriptionUpdate.scala

```diff
@@ -74,6 +75,43 @@ object ZuoraSubscriptionUpdate {
         }
     }
   }
+
+  def updateOfRatePlansToCurrentMembership2023(
+      subscription: ZuoraSubscription,
+      invoiceList: ZuoraInvoiceList,
+      effectiveDate: LocalDate,
+  ): Either[AmendmentDataFailure, ZuoraSubscriptionUpdate] = {
+
+    // This variant has a simpler signature than its classic counterpart.
+
+    val activeRatePlans = (for {
+      invoiceItem <- ZuoraInvoiceItem.items(invoiceList, subscription, effectiveDate)
+      ratePlanCharge <- ZuoraRatePlanCharge.matchingRatePlanCharge(subscription, invoiceItem).toSeq
+      price <- ratePlanCharge.price.toSeq
+      if price > 0
+      ratePlan <- ZuoraRatePlan.ratePlan(subscription, ratePlanCharge).toSeq
+    } yield ratePlan).distinct
+
+    if (activeRatePlans.isEmpty)
+      Left(AmendmentDataFailure(s"No rate plans to update for subscription ${subscription.subscriptionNumber}"))
+    else if (activeRatePlans.size > 1)
+      Left(AmendmentDataFailure(s"Multiple rate plans to update: ${activeRatePlans.map(_.id)}"))
+    else {
+
+      // At this point we know that we have exactly one activeRatePlans
+      val activeRatePlan = activeRatePlans.head
+
+      // In the case of Membership Batch 1, things are now more simple. We can hardcode the rate plan
+      Right(
+        ZuoraSubscriptionUpdate(
+          add = List(AddZuoraRatePlan("8a1287c586832d250186a2040b1548fe", effectiveDate)),
```

Comment:
Right, so we are not just updating the rate plan but migrating to a new one? Why was it erroring before though, did we decommission the current membership rate plan so it doesn't exist?

Reply:
It was erroring because the `ZuoraSubscriptionUpdate` object for membership had the new plan and the old plan mixed up. I haven't yet reversed engineered why that was the case (this will come when I start writing tests for the legacy code), I think that the legacy code had an assumption that is broken with membership